Add details of disciplinary action to CoC (#1438)
WebAssembly/design: CodeOfConduct.md

> vehicle for better identity of the organization.

We hope that our community group act according to these guidelines, and that
participants hold each other to these high standards. 

If you are worried that the code isn't being followed, or are witness to a violation but are
uncomfortable speaking up,please contact the Community Group's chairs at
`team-wasm-chairs@w3.org` (note: this list is also visible to W3C staff). 

In cases where there is repeated disregard for the CEPC, it is possible to incur disciplinary
action. The points below outline a series of steps that will be taken in case of violations
to ensure a positive work environment in the WebAssembly CG. 

 - **Informal warning**   

>    An informal or verbal warning can be issued in any relevant medium of communication callin 

Reviewer: ```suggestion
   An informal or verbal warning can be issued in any relevant medium of communication calling
```

   out inappropriate behavior as a violation of the CEPC
   (examples: responses to github issues or PRs, discord, email, meetings).

 - **Corrective action and/or counseling**   
   Informal meetings, or email discussions where problematic behavior is addressed,
   and suggestions for constructive ways forward are offered. A mediation process can also be an

>    option if relevant to the situation and both parties agree. 

Reviewer: Who is the second party here? The original reporter or possibly someone else?

Author: It depends on the the situation, it could be the original reporter, sometimes the reporter could be a different person than the one targeted, it could also be a relevant group of people.

 - **Formal warning**   
   An explicit warning issued after repeated incidents. While members of the CG can call out
   inappropriate behavior as needed, warnings will be issued by CG chairs, subgroup chairs, or

>    proposal champions depending on the nature of the violation.

Reviewer: What kind of reporting is expected from the chairs or champions giving out formal warnings? Can chairs and champions unilaterally give out formal warnings or is there some group that needs to be consulted first?

Author: This document is intentionally setup a a set of escalations, with persistent violations of the CEPC, I would expect that any formal warning if not publicly issued, will go through the chairs. I'll reword this point to be more clear.

 - **Disciplinary meeting**  

>    An optional meeting to discuss constructive action or next steps.

Reviewer: A meeting with whom?

Author: Currently this would be with the chairs and possibly with someone from the W3C if needed, in the future this could also take form of a committee. The details of that are not clear yet, but publishing this doc is an iterative step towards a more transparent process.

Reviewer 2: At whose option is the meeting? Maybe optional means that it can be requested by the person being warned (such that they get the opportunity to speak for themselves before chairs can escalate to suspension)? But if the person declines or is unresponsive, then they can go ahead?

Author: I've updated this a little bit, PTAL. Also cc @linclark to make sure this looks right.

> vehicle for better identity of the organization.

We hope that our community group act according to these guidelines, and that
participants hold each other to these high standards. 

If you are worried that the code isn't being followed, or are witness to a violation but are

> uncomfortable speaking up,please contact the Community Group's chairs at

Reviewer 3: Looks like there's a missing space here:

```suggestion
uncomfortable speaking up, please contact the Community Group's chairs at
```

Author: Thanks, fixed.

`team-wasm-chairs@w3.org` (note: this list is also visible to W3C staff). 

In cases where there is repeated disregard for the CEPC, it is possible to incur disciplinary
action. The points below outline a series of steps that will be taken in case of violations
to ensure a positive work environment in the WebAssembly CG. 

 - **Informal warning**   
   An informal or verbal warning can be issued in any relevant medium of communication calling
   out inappropriate behavior as a violation of the CEPC
   (examples: responses to github issues or PRs, discord, email, meetings).

 - **Corrective action and/or counseling**   
   Informal meetings, or email discussions where problematic behavior is addressed,
   and suggestions for constructive ways forward are offered. A mediation process can also be an
   option if relevant to the situation and both parties agree. 

 - **Formal warning**   
   An explicit warning issued after repeated incidents. While members of the CG can call out
   inappropriate behavior as needed, warnings will be issued by CG chairs, subgroup chairs, or

>    proposal champions depending on the nature of the violation. Wherever possible,

Reviewer 2: Do we want to make consultation with the chairs a requirement for a warning to be considered formal? It would of course be pretty surprising if there wasn't any chair involvement up to this point, but it seems like they should be at least aware.

Author: My original thought here was that this could sometimes be time sensitive, and enabling folks that are admins in repositories to issue warnings where needed makes sense to me. I've rephrased this some more to make the consultation a requirement, but explicit sign off is not required before a warning can be issued. Wdyt?

Reviewer 2: I think that makes sense. I think the current wording is good.

to ensure a positive work environment in the WebAssembly CG. 

 - **Informal warning**   
   An informal or verbal warning can be issued in any relevant medium of communication calling
   out inappropriate behavior as a violation of the CEPC
   (examples: responses to github issues or PRs, discord, email, meetings).

 - **Corrective action and/or counseling**   
   Informal meetings, or email discussions where problematic behavior is addressed,
   and suggestions for constructive ways forward are offered. A mediation process can also be an
   option if relevant to the situation and both parties agree. 

 - **Formal warning**   
   An explicit warning issued after repeated incidents. While members of the CG can call out
   inappropriate behavior as needed, warnings will be issued by CG chairs, subgroup chairs, or
   proposal champions depending on the nature of the violation. The warning should be issued
   with consultation from team-wasm-chairs@w3.org. 

 - **Optional formal meeting**  
   An optional meeting with the chairs and/or a W3C representative to discuss next steps.
   The meeting can be requested by any party, and should serve as a forum for discussion.

>    While this is the reccomended next step, it is not a blocker for further action, i.e.

Reviewer 2: ```suggestion
   While this is the recommended next step, it is not a blocker for further action, i.e.
```

>    the chairs can request thay a meeting take place, or should be available for a

Reviewer 2: ```suggestion
   the chairs can request that a meeting take place, and should be available for a
```

>    discussion if requested, but further action will not be gated on this meeting. 

Reviewer 2: ```suggestion
   discussion if requested by others, but further action will not be gated on this meeting. 
```
